src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationExtensions.cs

```diff
@@ -53,6 +54,37 @@ public static class DaprSecretStoreConfigurationExtensions
             return configurationBuilder;
         }
 
+        /// <summary>
+        /// Adds an <see cref="IConfigurationProvider"/> that reads configuration values from the Dapr Secret Store.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="store">Dapr secret store name.</param>
+        /// <param name="secretDescriptors">The secrets to retrieve.</param>
+        /// <param name="client">The Dapr client.</param>
+        /// <param name="sidecarWaitDuration">The <see cref="TimeSpan"/> used to configure the timeout waiting for Dapr.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddDaprSecretStore(
+            this IConfigurationBuilder configurationBuilder,
+            string store,
+            IEnumerable<DaprSecretDescriptor> secretDescriptors,
+            DaprClient client,
+            TimeSpan sidecarWaitDuration)
```

Comment:
Minor nitpick: you called this `...duration` but the docs say `... timeout`. Would be simpler to be consistent about the terms used.